concept/type/impl/TypeImpl.java

```diff
@@ -92,78 +145,83 @@ public final String getLabel() {
 
         @Override
         public final void setLabel(String label) {
-            final ConceptProto.TypeMethod.Req method = ConceptProto.TypeMethod.Req.newBuilder()
+            final TypeMethod.Req method = TypeMethod.Req.newBuilder()
                     .setTypeSetLabelReq(ConceptProto.Type.SetLabel.Req.newBuilder()
                                                 .setLabel(label)).build();
-            runMethod(method);
+            execute(method);
         }
 
         @Override
         public final boolean isAbstract() {
-            final ConceptProto.TypeMethod.Req method = ConceptProto.TypeMethod.Req.newBuilder()
+            final TypeMethod.Req method = TypeMethod.Req.newBuilder()
                     .setTypeIsAbstractReq(ConceptProto.Type.IsAbstract.Req.getDefaultInstance()).build();
 
-            return runMethod(method).getTypeIsAbstractRes().getAbstract();
+            return execute(method).getTypeIsAbstractRes().getAbstract();
+        }
+
+        void setSupertypeExecute(Type type) {
+            execute(TypeMethod.Req.newBuilder().setTypeSetSupertypeReq(SetSupertype.Req.newBuilder().setType(type(type))).build());
         }
 
         @Nullable
-        protected <TYPE extends Type.Remote> TYPE getSupertypeInternal(final Function<Type.Remote, TYPE> typeConverter) {
-            final ConceptProto.TypeMethod.Req method = ConceptProto.TypeMethod.Req.newBuilder()
+        <TYPE extends Type.Local> TYPE getSupertype(final Function<Type.Local, TYPE> typeConstructor) {
```

Comment:
The reason I named this `getSupertypeInternal` is because it has a different signature to `getSupertype` - this "internal" implementation detail takes in a `typeConstructor` function (to handle the case where the supertype is `null` elegantly).

Are we happy to give it the same name even though it's kind of different?

Reply:
I find that the overloading of `getSupertype(Function<T,U>)` vs `getSupertype()` was already a sufficient distinction, as we leverage Java's `overloading` feature. However, it did not work with `setSupertype(...)` as they take the same argument type. I don't feel too strongly about this. I see the reason to rename the method with a suffix again, and make them consistent. I am, however, against naming anything with "internal", as it does not describe _"what the method does"_, but rather _"where the method is"_ (which is not useful). I chose "*Execute" as it describes the method being the "executor", but maybe there's a better name.

Reply:
`execute` is fine. Renamed `getSupertype` in `TypeImpl` to `getSupertypeExecute`.